### PR TITLE
Made text blocks bookmarkable and created few extension points for easier customization

### PR DESCRIPTION
**Pull request by musketyr**
I'm testing using styles but it's not production ready so I'd rather created some overridable methods in the word builder so I can continue in my own subclass and push the changes when ready. I've let the code for generating styles commented out so you can see it.

### CODE REVIEW

File: word/src/main/groovy/com/craigburke/document/builder/BasicDocumentPartTypes.groovy [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package com.craigburke.document.builder
+
+/**
+ * Enum for the various types of document parts a relationship can exist in
+ * @author Craig Burke
```
Comment (craigburke) [inline word/src/main/groovy/com/craigburke/document/builder/BasicDocumentPartTypes.groovy:5]: I think you misspelled your name here ;)

**Comment by craigburke** [comment]
Looks really good, man. Do you happen to have an example of how the extension points would be used (like writeCustomFiles)?

**Comment by musketyr** [comment]
@craigburke see https://github.com/MetadataRegistry/ModelCataloguePlugin/blob/bcc9a674da2051a67fe20f0f202467c04eef56bf/ModelCatalogueGenomicsPlugin/src/groovy/org/modelcatalogue/core/gel/GelWordDocumentBuilder.groovy#L18 for writing custom files ('styles' and 'stylesWithEffects'). 

I'm trying to align the headings so latest MS Word can generate the TOC from it easily but with no luck (LibreOffice are bit better in this but it messes up the styles of generated TOC - all items appears to be on same level as the indentation is same there)

I've fixed the name :-)

**Comment by craigburke** [comment]
@musketyr I'm definitely not opposed to an extension point like this even if it's specific to the Word builder.

I'm wondering if some of the need for this extension point could be eliminated if I was just a bit smarter about how I handle template styles:  
http://www.craigburke.com/document-builder/#_templates_and_styles

Right now, setting the style for a paragraph just styles that text inline. I could instead use the template to generate a styles.xml instead. I'm not sure I fully understand everything going on in the PR yet so just let me know if I'm not making sense :smile:

**Comment by musketyr** [comment]
That's the next step. I would like to generate `styles.xml` file based on your templates.

To generate TOC you need to use proper styles not just styled paragraph, therefore I need to generate the `styles.xml` file.

I'm playing with the idea of using MS word template files directly (as that's the best DSL to describe styles there :smile:). That's why I've also added ability to generate any file in that `docx` file.

**Comment by craigburke** [comment]
Oh man, I really like where this is going. :+1: 

I want to take a better look at this and see if I can get us closer to that before doing another release (hopefully I'll be able to do that later in the week) :smile: 

Awesome awesome stuff!!!

**Comment by craigburke** [comment]
I merged your bookmark stuff (looks great). I removed the writeCustomFiles stuff although I'd like to revisit that part in 0.5.0 perhaps. I guess I have some work to do on the pdf side of things :smile: 

You ok, if I add you as a contributor in the docs?

**Comment by musketyr** [comment]
I have no problem with adding me to the contributors :-) I'm using my
custom build so no worries.

**Pull request merged**